src/rust/engine/src/nodes.rs

```diff
@@ -90,173 +90,83 @@ pub trait WrappedNode: Into<NodeKey> {
 }
 
 ///
-/// A Node that selects a product for a subject.
+/// A Node that selects a product for some Params.
 ///
 /// A Select can be satisfied by multiple sources, but fails if multiple sources produce a value.
-/// The 'variants' field represents variant configuration that is propagated to dependencies. When
-/// a task needs to consume a product as configured by the variants map, it can pass variant_key,
-/// which matches a 'variant' value to restrict the names of values selected by a SelectNode.
+/// The 'params' represent a series of type-keyed parameters that will be used by Nodes in the
+/// subgraph below this Select.
 ///
 #[derive(Clone, Debug, Eq, Hash, PartialEq)]
 pub struct Select {
-  pub subject: Key,
-  pub variants: Variants,
+  pub params: Params,
   pub selector: selectors::Select,
-  entries: rule_graph::Entries,
+  entry: rule_graph::Entry,
 }
 
 impl Select {
-  pub fn new(
-    product: TypeConstraint,
-    subject: Key,
-    variants: Variants,
-    edges: &rule_graph::RuleEdges,
-  ) -> Select {
-    let selector = selectors::Select::without_variant(product);
-    let select_key = rule_graph::SelectKey::JustSelect(selector.clone());
-    Select {
-      selector: selector,
-      subject: subject,
-      variants: variants,
-      entries: edges.entries_for(&select_key),
-    }
+  pub fn new(product: TypeConstraint, params: Params, edges: &rule_graph::RuleEdges) -> Select {
+    Self::new_with_selector(selectors::Select::without_variant(product), params, edges)
```

Comment:
Is the plan for `Select` to disappear in favour of just a `TypeConstraint` (or maybe a wrapper-type around it), and for `tasks_add_select_variant` to disappear?